gui/default/syncthing/core/syncthingController.js

```diff
@@ -1603,20 +1604,23 @@ angular.module('syncthing.core')
                 // here as well...
                 $scope.devices = deviceMap($scope.config.devices);
 
-                $scope.saveConfig().then(function () {
-                    if (themeChanged) {
-                        document.location.reload(true);
-                    }
+                $('#settings').one('hidden.modal.bs', function () {
+                    $scope.saveConfig().then(function () {
```

Comment:
Is there an advantage to delaying saving the config until the modal has closed?

Reply:
This is the essence of this PR actually. Without delaying the config save, the "Saving Changes" modal overlaps with the already open Settings modal, which adds padding to the body that doesn't get removed even after closing the modals later.

Another advantage is that saving changes only after the modal has been closed makes it 100% certain that the user hasn't added any last-minute modifications while the saving was already in process (but the modal still stayed open), because otherwise such modifications would be lost.